### PR TITLE
users: Rename locking in create new account

### PR DESCRIPTION
**Pull request by tomasmatus**
Fixes: #16886

Edit: screenshots

account_menu_fix2
account_menu_popover_fix2
create_new_acc_fix2
create_new_acc_popover_fix2

### CODE REVIEW

**Comment by marusak** [comment]
Can you please post screenshot how this looks like?

**Comment by tomasmatus** [comment]
Somehow I managed to link incorrect issue and changed the text to "Disallow **account** usage" in the first screenshot
Both will be fixed tomorrow

**Review by garrett — CHANGES_REQUESTED** [review]
Cool! And thanks for the screenshots too!

However:

- The popover talks about it currently being disabled. We don't want two variants of the explanation for when it's checked and when it's not. It's better to explain the reason why without including the current state.
- The (?) icon should go to the right of the checkbox's text label, not next to options. (It should _not_ be part of the text label. Clicking on (?) should definitely _not_ toggle the checkbox.)

The popover text could probably be something like:

"Other authentication methods are still available even when password authentication is disallowed."

After you fix these issues, it's probably good to merge from my viewpoint.

**Comment by tomasmatus** [comment]
I moved the (?) icon behind checkbox and changed popup message like you said.
Please have a look on new screenshots above

File: pkg/users/account-create-dialog.js [hunk 1]
```diff
@@ -66,10 +67,20 @@ function AccountCreateBody({ state, errors, change }) {
...
-            <FormGroup label={_("Access")} fieldId="accounts-create-locked" isInline>
+            <FormGroup label={_("Options")} fieldId="accounts-create-locked" hasNoPaddingTop>
                 <Checkbox id="accounts-create-locked"
-                          label={_("Lock account")}
-                          isChecked={locked} onChange={checked => change("locked", checked)} />
+                          isChecked={locked} onChange={checked => change("locked", checked)}
+                          label={
+                              <>
+                                  {_("Disallow password usage")}
+                                  <Popover bodyContent={_("Other authentication methods are still available even when password authentication is disallowed")}>
```
Comment (garrett) [inline pkg/users/account-create-dialog.js:76]:
```suggestion
                                  <Popover bodyContent={_("Other authentication methods are still available even when password authentication is disallowed.")}>
```

This popover has a sentence. It needs a period.

File: pkg/users/account-details.js [hunk 2]
```diff
@@ -312,14 +312,24 @@ export function AccountDetails({ accounts, groups, shadow, current_user, user })
...
                                             <Checkbox id="account-locked"
                                                       isDisabled={!superuser.allowed || edited_locked != null || user == current_user}
                                                       isChecked={edited_locked != null ? edited_locked : details.locked}
-                                                      label={_("Lock account")}
-                                                      onChange={checked => change_locked(checked)} />
+                                                      onChange={checked => change_locked(checked)}
+                                                      label={
+                                                          <>
+                                                              {_("Disallow password usage")}
+                                                              <Popover bodyContent={_("Other authentication methods are still available even when password authentication is disallowed")}>
```
Comment (garrett) [inline pkg/users/account-details.js:325]:
```suggestion
                                                              <Popover bodyContent={_("Other authentication methods are still available even when interactive password authentication is not allowed.")}>
```

Should

File: pkg/users/account-create-dialog.js [hunk 3]
```diff
@@ -66,10 +67,20 @@ function AccountCreateBody({ state, errors, change }) {
...
 
-            <FormGroup label={_("Access")} fieldId="accounts-create-locked" isInline>
+            <FormGroup label={_("Options")} fieldId="accounts-create-locked" hasNoPaddingTop>
                 <Checkbox id="accounts-create-locked"
-                          label={_("Lock account")}
-                          isChecked={locked} onChange={checked => change("locked", checked)} />
+                          isChecked={locked} onChange={checked => change("locked", checked)}
+                          label={
+                              <>
+                                  {_("Disallow password usage")}
```
Comment (garrett) [inline pkg/users/account-create-dialog.js:75]:
```suggestion
                                  {_("Disallow interactive password")}
```

File: pkg/users/account-details.js [hunk 4]
```diff
@@ -312,14 +312,24 @@ export function AccountDetails({ accounts, groups, shadow, current_user, user })
...
                                         <div className="account-column-one">
                                             <Checkbox id="account-locked"
                                                       isDisabled={!superuser.allowed || edited_locked != null || user == current_user}
                                                       isChecked={edited_locked != null ? edited_locked : details.locked}
-                                                      label={_("Lock account")}
-                                                      onChange={checked => change_locked(checked)} />
+                                                      onChange={checked => change_locked(checked)}
+                                                      label={
+                                                          <>
+                                                              {_("Disallow password usage")}
```
Comment (garrett) [inline pkg/users/account-details.js:324]:
```suggestion
                                                              {_("Disallow interactive password")}
```

**Review by garrett — CHANGES_REQUESTED** [review]
This looks great!

Apologies for taking a while to get back to this.

I was looking at it again and some clearer text occurred to me. I'll add suggestions inline. Other than the text adjustments, it looks great though! (Sorry for changing things.)

**Comment by tomasmatus** [comment]
No worries, this definitely sounds more polished.
I updated screenshots above.

**Comment by marusak** [comment]
I think that [users: Rename locking in create new account](https://github.com/cockpit-project/cockpit/pull/16908/commits/d6bd4741165b4a0871058b1df40c34ccb0522907) and [users: Rename locking in create new account](https://github.com/cockpit-project/cockpit/pull/16908/commits/d6bd4741165b4a0871058b1df40c34ccb0522907) commits could be squashed. Can you please take a look?

File: pkg/users/account-create-dialog.js [hunk 5]
```diff
@@ -66,10 +67,20 @@ function AccountCreateBody({ state, errors, change }) {
...
+                          label={
+                              <>
+                                  {_("Disallow interactive password")}
+                                  <Popover bodyContent={_("Other authentication methods are still available even when interactive password authentication is not allowed.")}>
+                                      <button onClick={e => e.preventDefault()}
+                                      className="pf-c-form__group-label-help">
+                                          <HelpIcon noVerticalAlign />
+                                      </button>
+                                  </Popover>
+                              </>
```
[the remark this answers is not in the document]
Comment (jelly) [reply pkg/users/account-create-dialog.js:82]: If this is an issue , we need to change it here as well? https://github.com/cockpit-project/cockpit/blob/ba8700a6e501a9539b7a72d1d395649a8263cb69/pkg/users/account-roles.js#L74

[the remark this answers is not in the document]
Comment (marusak) [reply pkg/users/account-create-dialog.js]: I think so. Now when you click on the (?) it toggles the checkbox.

[the remark this answers is not in the document]
Comment (garrett) [reply pkg/users/account-create-dialog.js]: Correct. Having the icon in the label causes three issues:

1. Clicking on the (?) toggles the checkbox (and it should not).
2. Screen readers interpret the action incorrectly.
3. Labels should not have children that are widgets (such as buttons), so it's invalid HTML.
   - See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#accessibility_concerns

[the remark this answers is not in the document]
Comment (tomasmatus) [reply pkg/users/account-create-dialog.js]: OK @garrett  ! Moved the popover out of label and fixed the issue jelly mentioned.
I also updated screenshots.

**Review by garrett — APPROVED** [review]
LGTM (Looks Good To Me)! :+1: 

---

I did notice a weird error that does talk about unlocking and is therefore semi-related:

Screenshot 2022-02-15 at 12-54-52 Accounts - garrett Bolt

I encountered this by unchecking the disallow password setting.

We probably want to change the text... and also perhaps make it disabled for a scenario like this?

Could we address it in a follow-up?

**Comment by marusak** [comment]
> We probably want to change the text

I don't think that is our test, I guess it comes from `usermod` utlity

>  perhaps make it disabled for a scenario like this

If we can recognize this case, then yes. Can you please post reproducer for this? (what kind of account is created...)

>  follow-up?

yes, this PR is getting big on its own already :)

**Pull request merged**